tc/tc_util.c

```diff
@@ -192,6 +192,86 @@ static const struct rate_suffix {
 };
 
 
+int get_percent_rate(unsigned int *rate, const char *str, char *dev)
+{
+	long value;
+	char percent[strlen(dev) - 1], r_str[20];
+	const char *pos;
+	int ret;
+	double d, r;
+
+	if (read_prop(dev, "speed", &value))
+		return -1;
+
+	pos = str + strlen(str) -1;
+	if (*pos != '%') {
+		fprintf(stderr, "Specified rate value is malformed \n");
+		return -1;
+	}
+
+	strncpy(percent, str, strlen(str) -1);
+	ret = sscanf(percent, "%lf", &d);
+	if (ret != 1) {
+		fprintf(stderr, "Specified rate value is malformed \n");
```

Comment:
Replace the error statement with this one:

    fprintf(stderr, "Specified rate value could not be read or is malformed: %s\n", str);